daemon/daemon.go

```diff
@@ -448,6 +448,11 @@ func (daemon *Daemon) registerLink(parent, child *container.Container, alias str
 
 // SetClusterProvider sets a component for querying the current cluster state.
 func (daemon *Daemon) SetClusterProvider(clusterProvider cluster.Provider) {
+	// Daemon is in charge of removing the attachable networks with
+	// connected containers when the node leaves the swarm
+	if clusterProvider == nil {
+		daemon.clearAttachableNetworks()
+	}
```

Comment:
That may be true now, but what happens when the code changes?

I still think this is extremely misleading behaviour for a function called `SetClusterProvider()`. I would much rather have a few lines duplicated next to each other (clear then set), instead of having a set method perform some kind of clean up on unrelated objects.